plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeBucketFunction.java

```diff
@@ -37,7 +37,7 @@
     public DeltaLakeBucketFunction(TypeOperators typeOperators, List<DeltaLakeColumnHandle> partitioningColumns, int bucketCount)
     {
         this.hashCodeInvokers = partitioningColumns.stream()
-                .map(DeltaLakeColumnHandle::getType)
+                .map(DeltaLakeColumnHandle::getBaseType)
```

Comment:
this should use the leaf type, not base type

did you intend to verify the column handle does not represent a dereference?